classes/controller/rest.php

```diff
@@ -64,6 +64,9 @@ public function before()
 		parent::before();
 
 		// Some Methods cant have a body
+		if (!$this->request) {
```

Comment:
You should use `empty` (or maybe `=== null`).

Reply:
or `is_object()` / `is_a()` / `instanceof` ?